### DISCUSSION TITLE
Wyze Camv3 - Problems with SSH and Go2RTC

### DISCUSSION THREAD

**Discussion by Tom8867** [gtxaspec/wz_mini_hacks#697] [Q&A]
Before I describe my problems, first the info what equipment I use and what I configured.
Before I switched to Wyze I used the old Xiaomi cubes with Fanghacks but one died so I bought 2 new Wyze Camv3.
I use them as surveillance cameras connected to my Synology NAS DS-418.
Firmware is v4.36.9.139 and running wz_mini 2024-01-01.
So far camera is working fine and recording works when I use v4l2rtspserver.
What I can't get to work properly is Go2RTC.
I installed on my Synology [Go2RTC from the Synology Community](https://synocommunity.com/package/go2rtc), it worked when I changed the IP address to [ip-address:8554/1080p].
But, the recording has dropouts, gets blurry, parts are distorted and worst is, Go2RTC ignores my settings of continuous recording of 30 minutes.
The recording varies between 2 and 3 minutes and starts a new recording after that.
v4l2rtspserver has no problems with that, also no dropouts or blurry parts or part of the videos look distorted.
Unfortunately v4l2rtspserver is now deprecated, so no choice to switch sooner or later to Go2RTC.
But the documentation is underwhelming for this specific use case, the Go2RTC plugin has no configuration options, also no documentation about that and the Go2RTC Github is not covering this configuration.
What can I do that Go2RTC respects the settings from my NAS and have a smooth recording from my webcams?
Second problem is SSH.
I think I read everything (incl. links) in Issues and here in discussion, but every single time I want to connect to the camera, I get a connection refused (WinSCP), Putty doesn't find my public and private key and asks me to generate new ones and Windows Terminal gives also an error message.
I also read the topic about dropbear, but no go to use this under Windows (and I believe I googled everything.
Keys were generated in Windows Terminal with the command "ssh-keygen -m ED25519" and the pub key was written into the "authorized_keys" with Notepad++. ED25519 seems to be the one which was discussed in the topic of about dropbear.
Best would be really, if I could disable the whole SSH for the Wyze Cam, but I haven't seen an option for that, or could access all configurations by the Webserver.
Anyone has some further ideas? 
If any information is missing, I can provide this of course.

Cheers
Tom

**Comment by archandanime** [comment]
I don't have much experience with Go2RTC but believe I can assist you with dropbear.

As far as I know public key type should not affect the connection as long as your SSH key is either `rsa`, `dss`, `ecdsa` or `ed25519`(I am using `rsa` on Linux now)

------

Here's my attempt to connect SSH using Windows 11 Terminal:

Generate SSH key:

```
PS C:\Users\Thinkpad\.ssh> ssh-keygen -t ed25519
Generating public/private ed25519 key pair.
Enter file in which to save the key (C:\Users\Thinkpad/.ssh/id_ed25519):
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Your identification has been saved in C:\Users\Thinkpad/.ssh/id_ed25519
Your public key has been saved in C:\Users\Thinkpad/.ssh/id_ed25519.pub
The key fingerprint is:
SHA256:uGs90Zd2dSaYraBcEEPAjvpIg2UgNBY5W8omOmROeqI thinkpad@DESKTOP
The key's randomart image is:
+--[ED25519 256]--+
|.=o  ..o+.       |
|+o..  . ..       |
|o.=  o   .   +   |
|.Oo . ..  o o o +|
|X+ .  ..S+ . o +.|
|*.=    .+ . = .  |
|.= +  .. . o .   |
|E . . ..o        |
|     ..  .       |
+----[SHA256]-----+
PS C:\Users\Thinkpad\.ssh> ls


    Directory: C:\Users\Thinkpad\.ssh


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----      Wed 3 1 2024   9:54 am            419 id_ed25519
-a----      Wed 3 1 2024   9:54 am            107 id_ed25519.pub
-a----      Wed 3 1 2024   9:52 am            193 known_hosts
PS C:\Users\Thinkpad> ssh root@192.168.10.20
root@192.168.10.20: Permission denied (publickey).
```

Then I added the public key (in `.ssh/id_ed25519.pub`) to `wz_mini/etc/ssh/authorized_keys` and tried to establish SSH connection:
```
PS C:\Users\Thinkpad> ssh root@192.168.10.20
COLUMNS=120;LINES=30;export COLUMNS LINES;
[root@WCV3:~]# 
```
 
It worked :)

**Reply by Tom8867** [reply]
Thank you so much, but this is exactly what I did and didn't work for me. 
Error message:
`The authenticity of host '192.168.1.59 (192.168.1.59)' can't be established.
ED25519 key fingerprint is SHA256:8hmwfn0QFSEMSbobebsLQ9fWOBSxxxxxxxxx.
This key is not known by any other names
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Warning: Permanently added '192.168.1.59' (ED25519) to the list of known hosts.
root@192.168.1.59: Permission denied (publickey).`
I wonder now whether this is a formatting problem when I add the pub key to authorized_keys with Notepad++.

Cheers
Tom

**Reply by archandanime** [reply]
It seems that something is wrong with the `authorized_keys` file, would you mind double checking if the public key has been added correctly into the file? Ah, also don't forget to use Notepad with Unix line ending(LF), it's on the bottom of the text editor window.

**Reply by Tom8867** [reply]
That was it, changed in Notepad++ to Unix Line ending(LF), it used the Windows (CR/LF).
Deleted the know_hosts file and it worked.
Now only the Go2RTC is the remaining problem, but seems I am not the only one who faces this issue.

Cheers
Tom

**Comment by MaxxyM13** [comment]
@Tom8867 , how did you get [Go2RTC from the Synology Community](https://synocommunity.com/package/go2rtc) working? I tried to update RTSP listening address in go2rtc.yaml to no avail. And I couldn't figure out how to specify the full URL including the path...

**Reply by Tom8867** [reply]
I donwloaded it and installed it, didn't see any configuration options and really no idea what it is doing in the background.
Also, there is no documentation and it wasn't mentioned in the forum of the community. 
Before that, when I tested Go2RTC, I had maybe a stream for max. 2 minutes in the Synology Surveillance Center and then it stopped incl. recording. 
After the installation, the stream didn't freeze (which is already an improvement) but still the problems I described above.

I didn't make any changes to the go2rtc.yaml and used the following URL: rtsp://user:pass@ip address::8554/1080p. 
Previously I used port 1984, but this didn't work correctly
For the installation I used 
Brand: user defined
Type: Streaming - RTSP.

Cheers
Tom

**Reply by Bob-AOL** [reply]
The only option is to either activate it or not via the Wyze app at the very bottom of the Advanced Settings after flashing the firmware. When activated it (RTSP) simply provides a solid RTSP stream from the camera available on port 554 using the camera's authentication credentials and the MRL provided. It really is that simple. When all else fails, and though many prefer to disregard it, the Wyze app can actually be your friend at times.

And the wz_mini RTSP options are still functional with the exception being the server itself and the corresponding MRL (URL). That said, it will also work as free standing Wyze firmware without wz_mini. 

If you can't get it to work in either scenario, I would be looking in a different direction for a solution to your difficulties.

**Comment by Bob-AOL** [comment]
I use the 4.61.0.1/3 Wyze Beta RTSP firmware with an earlier wz_mini distro and experience no problems whatsoever with streaming through and recording to Blue Iris or my Amcrest XVRs. It does become the dominant RTSP server over the wz_mini RTSP server in this scenario, but works just fine for my particular application otherwise. Just establish your credentials via the Wyze app and go. Doesn't get much simpler. rtsp://user:pass@ip address:554/live

Some say the Wyze RTSP firmware is junk. I certainly don't know why and would wholeheartedly disagree. I run it on eight v3s, both wired and wireless with zero performance issues.
